main.go

```diff
@@ -92,22 +109,71 @@ func printVersion() {
 }
 
 var (
-	errNoPackageFound = errors.New("no package found")
+	errNoPackageFound        = errors.New("no package found")
+	errUnsupportMultiplePkgs = errors.New(
+		`multiple packages are not supported, 
+		please specify the package ID or directory path that only contains one package`,
+	)
+	errMissingPackageName = errors.New("missing package name")
 )
 
 func generateFromRequest(request GenerationRequest) (string, error) {
 	cfg := packages.Config{
 		Mode: packages.NeedName | packages.NeedTypes,
 	}
-	pkgs, err := packages.Load(&cfg, request.Pkg)
+	if request.ModelPkg == "" {
+		request.ModelPkg = request.Pkg
+	}
+	if request.DestPkg == "" {
+		request.DestPkg = request.Pkg
+	}
+	intfPkgID, err := getPkgID(request.Pkg)
 	if err != nil {
 		return "", err
 	}
-	if len(pkgs) == 0 {
-		return "", errNoPackageFound
+	modelPkgID, err := getPkgID(request.ModelPkg)
+	if err != nil {
+		return "", err
+	}
+	destPkgID, err := getPkgID(request.DestPkg)
+	if err != nil {
+		return "", err
+	}
+	pkgs, err := packages.Load(&cfg, intfPkgID, modelPkgID, destPkgID)
+	if err != nil {
+		return "", err
 	}
+	pkgM := packagesToMap(pkgs)
+	return generator.GenerateRepositoryImpl(
+		pkgM[modelPkgID].Types,
+		pkgM[intfPkgID].Types,
+		pkgM[destPkgID].Types,
+		request.ModelName,
+		request.RepoName,
+	)
+}
 
-	pkg := pkgs[0]
+func getPkgID(pattern string) (string, error) {
+	pkgs, err := packages.Load(nil, pattern)
+	if err != nil {
+		return "", err
+	}
+	if len(pkgs) < 1 {
+		return "", errNoPackageFound
+	}
+	if len(pkgs) > 1 {
+		return "", errUnsupportMultiplePkgs
+	}
+	if pkgs[0].Name == "" {
+		return "", fmt.Errorf("%w on %s", errMissingPackageName, pattern)
```

Comment:
I found that when given path is an empty folder, it will still return a package with empty `Name` field ..

Reply:
That's interesting... You may comment above the if statement to explain why this check exists.